### PR TITLE
Allow MoneyField to be used without a model serializer

### LINKED ISSUE #757: Regression in 3.5 - AttributeError: type object 'Meta' has no attribute 'model'
Looks to be caused by https://github.com/django-money/django-money/pull/722.

Django - 4.2.11
DRF - 3.14.0

```
  File "/Users/phillip/Library/Caches/pypoetry/virtualenvs/shipwell-backend-core-cpwchxt0-py3.11/lib/python3.11/site-packages/rest_framework/serializers.py", line 426, in run_validation
    value = self.to_internal_value(data)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/phillip/Library/Caches/pypoetry/virtualenvs/shipwell-backend-core-cpwchxt0-py3.11/lib/python3.11/site-packages/rest_framework/serializers.py", line 481, in to_internal_value
    primitive_value = field.get_value(data)
                      ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/phillip/Library/Caches/pypoetry/virtualenvs/shipwell-backend-core-cpwchxt0-py3.11/lib/python3.11/site-packages/djmoney/contrib/django_rest_framework/fields.py", line 70, in get_value
    model_meta = self.parent.Meta.model._meta
                 ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: type object 'Meta' has no attribute 'model'
```

I think this is an edge case when using a `MoneyField` with a `Meta` and outside of a `ModelSerializer`. `ModelSerializer` is the only case where a `model` is used in DRF.

This should be the reproduction case:

```python
class NonModelSerializer(serializers.Serializer):
    invoice_amount = MoneyField(
        max_digits=constants.DIGIT_SIZE_10,
        decimal_places=constants.DIGIT_SIZE_2,
        required=False,
        allow_null=True,
    )

    class Meta:
        fields = (
            "invoice_amount"
        )
```

### PR DESCRIPTION
**Pull request by phillipuniverse**
Fixes #757 from original regression caused by #722.

I added a test but I also manually validated in my environment and this fixes the stack trace I was getting!

### CODE REVIEW

File: tests/contrib/test_django_rest_framework.py [hunk 1]
```diff
@@ -261,3 +261,15 @@ class Meta:
         else:
             assert not serializer.is_valid()
             assert serializer.errors["second_money"][0] == "Ensure this value is greater than or equal to 0."
+
+    def test_no_model_serializer(self):
```
Comment (phillipuniverse) [inline tests/contrib/test_django_rest_framework.py:265]: Confirmed this test failed prior to my fix

Comment (benjaoming) [reply tests/contrib/test_django_rest_framework.py:265]: :pray:

**Review by benjaoming — APPROVED** [review]
Thanks for the fix + test case :100:

**Comment by benjaoming** [comment]
Thanks for this! It's been released in 3.5.1

**Pull request merged**